WordPress/src/main/java/org/wordpress/android/ui/main/SitePickerAdapter.java

```diff
@@ -541,16 +541,22 @@ public int compare(SiteRecord site1, SiteRecord site2) {
             }
 
             if (mSites == null || !mSites.isSameList(sites)) {
-                mAllSites = (SiteList) sites.clone();
-                mSites = filteredSitesByTextIfInSearchMode(sites);
+                SiteList allSites = (SiteList) sites.clone();
+                SiteList filteredSites = filteredSitesByTextIfInSearchMode(sites);
+
+                return new SiteList[]{allSites, filteredSites};
             }
 
             return null;
         }
 
         @Override
-        protected void onPostExecute(Void results) {
-            notifyDataSetChanged();
+        protected void onPostExecute(SiteList[] updatedSiteLists) {
+            if (updatedSiteLists != null) {
```

Comment:
> Theory:
>
> > The fix tries to put 

I mean some specific scenario, like perhaps "Background thread <X> may use the updated mSites and because `notifyDataSetChanged()` is not called yet, ...".

I can't envision a scenario like that in this Adapter so far.